digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/CsvDataExporter.scala

```diff
@@ -0,0 +1,108 @@
+package fi.liikennevirasto.digiroad2
+
+import fi.liikennevirasto.digiroad2.dao.ExportReportDAO
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+import org.joda.time.DateTime
+
+
+case class ExportStatusInfo(id: Long, status: Int, statusDescription: String, fileName: String,
+                            createdBy: Option[String], createdDate: Option[DateTime], exportedAssets: String,
+                            municipalities: String, content: Option[String])
+
+
+trait CsvDataExporterOperations {
+
+    def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+    def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+    def eventBus: DigiroadEventBus
+    val exportReportDAO: ExportReportDAO = new ExportReportDAO
+
+    def insertData(username: String, fileName: String, exportType: String, municipalities: String, withTransaction: Boolean = true): Long = {
+      if (withTransaction)
+        withDynTransaction {
+          exportReportDAO.create(username, fileName, exportType, municipalities )
+        }
+      else
+        exportReportDAO.create(username, fileName, exportType, municipalities )
+
+    }
+
+    def getExportById(id: Long, withTransaction: Boolean = true) : Option[ExportStatusInfo]  = {
+      if (withTransaction)
+        withDynTransaction {
+          exportReportDAO.get(id)
+        }
+      else
+        exportReportDAO.get(id)
+    }
+
+    def getByUser(username: String, withTransaction: Boolean = true) : Seq[ExportStatusInfo]  = {
+      if (withTransaction)
+        withDynTransaction {
+          exportReportDAO.getByUser(username)
+        }
+      else
+        exportReportDAO.getByUser(username)
+    }
+
+    def getById(id: Long, withTransaction: Boolean = true) : Option[ExportStatusInfo]  = {
+      if (withTransaction)
+        withDynTransaction {
+          exportReportDAO.get(id)
+        }
+      else
+        exportReportDAO.get(id)
+    }
+
+    def getByIds(ids: Set[Long], withTransaction: Boolean = true) : Seq[ExportStatusInfo]  = {
+      if (withTransaction)
+        withDynTransaction {
+          exportReportDAO.getByIds(ids)
+        }
+      else
+        exportReportDAO.getByIds(ids)
+    }
+
+    def update(id: Long, status: Status, content: Option[String] = None, withTransaction: Boolean = true) : Long  = {
+      if (withTransaction)
+        withDynTransaction {
+          exportReportDAO.update(id, status, content)
+        }
+      else
+        exportReportDAO.update(id, status, content)
+    }
+
+    def createStandardCSVData(headers: Seq[String], values: Seq[Map[String, String]]): String = {
```

Comment:
Is this function used only in the Spec? Then we should move it to that in my opinion

Reply:
Yes.
This function is to create a standard CSV export.
We don't have for now a export that is a standard csv, but the code is here to be ready in future.
However it maybe require some adjustments

Reply:
That's true, I remember now we discussed that, my bad.